jetty-cdi/src/main/java/org/eclipse/jetty/cdi/CdiServletContainerInitializer.java

```diff
@@ -0,0 +1,63 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.cdi;
+
+import java.util.Set;
+import javax.servlet.ServletContainerInitializer;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+
+import org.eclipse.jetty.annotations.AnnotationConfiguration;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.webapp.WebAppContext;
+
+/**
+ * A {@link ServletContainerInitializer} that introspects for a CDI API
+ * implementation within a web application. If the CDI API is found, then
+ * a {@link CdiDecorator} is registered as a {@link org.eclipse.jetty.util.Decorator}
+ * for the context.
+ * @see AnnotationConfiguration.ServletContainerInitializerOrdering
+ */
+public class CdiServletContainerInitializer implements ServletContainerInitializer
+{
+    public static final String CDI_INTEGRATION_ATTRIBUTE = "org.eclipse.jetty.cdi";
+    private static final Logger LOG = Log.getLogger(CdiServletContainerInitializer.class);
+
+    @Override
+    public void onStartup(Set<Class<?>> c, ServletContext ctx) throws ServletException
+    {
+        try
+        {
+            WebAppContext context = WebAppContext.getCurrentWebAppContext();
+            context.getObjectFactory().addDecorator(new CdiDecorator(context));
+            context.setAttribute(CDI_INTEGRATION_ATTRIBUTE, "CdiDecorator");
```

Comment:
I understand this is a "mode" that CDI application should use to test what to do, right?
The modes are "decorator" or "listener", right?
If that's the case, then the value of the attribute should be "decorator" not "CdiDecorator" - it's a mode not a class.

Reply:
Kind off.... except that the `decorate` module is not CDI specific, so it is not only a CDI integration mode, so can't have the one attribute that switches for them both.

So there is an attribute to say what attribute the DecoratingListener is listening for, and another attribute to say what the CDI integration mode is, which currently can only be "CdiDecorator" or nothing.

Reply:
Too late to change the attribute name as weld has already released against this.